esm/encode.js

```diff
@@ -11,8 +13,9 @@ export const encode = async (buffer, format = 'deflate') => {
     response = response.stream().pipeThrough(stream);
   }
   const view = new Uint8Array(await new Response(response).arrayBuffer());
-  const out = [];
-  for (let i = 0; i < view.length; i += 2000)
-    out.push(String.fromCharCode(...view.subarray(i, i + 2000)));
-  return btoa(out.join(''));
+  let out = '';
+  const viewLength = view.length;
+  for (let i = 0; i < viewLength; i += VIEW_CHUNK_SIZE)
```

Comment:
`let cs = 2000, i = 0; i < …` this would be better, although it performs worse once compressed but it feels more reasonable